data/__init__.py

```diff
@@ -1,213 +1,75 @@
-            timestamp, variable)
-
-        if not file_list:
-            raise RuntimeError("NetCDF file list is empty.")
-
-        return file_list
-
-
-def __get_grid_angle_file_url(datasetconfig) -> str:
-    return datasetconfig.grid_angle_file_url
-
-
-def __get_requested_timestamps(db: SQLiteDatabase, variable: str, timestamp, endtime, nearest_timestamp) -> List[int]:
-
-    # We assume timestamp and/or endtime have been converted
-    # to the same time units as the requested dataset. Otherwise
-    # this won't work.
-    if nearest_timestamp:
-        all_timestamps = db.get_timestamps(variable)
-
-        start = find_le(all_timestamps, timestamp)
-        if not endtime:
-            return [start]
-
-        end = find_le(all_timestamps, endtime)
-        return db.get_timestamp_range(start, end, variable)
-
-    if timestamp > 0 and endtime is None:
-        # We've received a specific timestamp (e.g. 21100345)
-        if not isinstance(timestamp, list):
-            return [timestamp]
-        return timestamp
-
-    if timestamp < 0 and endtime is None:
-        all_timestamps = db.get_timestamps(variable)
-        return [all_timestamps[timestamp]]
-
-    if timestamp > 0 and endtime > 0:
-        # We've received a request for a time range
-        # with specific timestamps given
-        return db.get_timestamp_range(
-            timestamp, endtime, variable)
-
-    # Otherwise assume negative values are indices into timestamp list
-    all_timestamps = db.get_timestamps(variable)
-    len_timestamps = len(all_timestamps)
-    if timestamp < 0 and endtime > 0:
-        idx = roll_time(timestamp, len_timestamps)
-        return db.get_timestamp_range(all_timestamps[idx], endtime, variable)
-
-    if timestamp > 0 and endtime < 0:
-        idx = roll_time(endtime, len_timestamps)
-        return db.get_timestamp_range(timestamp, all_timestamps[idx], variable)
-
-
-def __is_mercator(dimension_list: list) -> bool:
-    return 'longitude' in dimension_list or 'latitude' in dimension_list
-
-
-def __is_fvcom(dimension_list: list) -> bool:
-    return 'siglay' in dimension_list
-
-
-def __get_dimensions(url: str) -> List[str]:
-
-    dimension_list = []
-
-    if __is_sqlite_database(url):
-
-        with SQLiteDatabase(url) as db:
-            dimension_list = db.get_all_dimensions()
-
-    elif __is_aggregated_or_raw_netcdf(url):
-        # Open dataset (can't use xarray here since it doesn't like FVCOM files)
-        ds = Dataset(url, 'r')
-
-        dimension_list = [dim for dim in ds.dimensions]
-
-        ds.close()
-
-    return dimension_list
+        calculated_vars = {}
+
+    if url is None:
+        raise ValueError("Dataset url is None.")
+
+    args = {
+        "calculated": calculated_vars,
+        "meta_only": kwargs.get("meta_only", False),
+        "grid_angle_file_url": getattr(dataset, "grid_angle_file_url", ""),
+    }
+    try:
+        args["dataset_key"] = dataset.key
```

Comment:
Good catch! Except I think it should be `"dataset_key": getattr(dataset, "key", "")` because `DatasetConfig` constructor says `self._dataset_key: str = dataset`.

Done in 6b974df5